### PR TITLE
[tests-only][full-ci] fix check for deleted suites in expected failure file

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
Fixes the check for deleted files in the expected failure files in case of typos.
- typos in suite name
- typos in feature file name

### Typo in expected failure file
```md
- [oreApiVersions/fileVersionAuthor.feature:14](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/coreApiVersions/fileVersionAuthor.feature#L14)
- [coreApiVersions/fileersionAuthor.feature:60](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/coreApiVersions/fileVersionAuthor.feature#L60)
- [coreApiVersions/fileVersionAuthor.feature:92](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/coreApiVersions/fileVersionAuthor.feature#L92)
```

### Outcome
```bash
The following test scenarios do not exist anymore:
They can be deleted from the 'expected-failures-API-on-OCIS-storage.md'
oreApiVersions/fileVersionAuthor.feature:14
coreApiVersions/fileersionAuthor.feature:60
```

## Related Issue

- Fixes https://github.com/owncloud/ocis/issues/10676

### CODE REVIEW

File: tests/acceptance/check-deleted-suites-in-expected-failure.sh [hunk 1]
```diff
@@ -44,9 +44,17 @@ for expected_failure_file in "${EXPECTED_FAILURE_FILES[@]}"; do
...
+    pattern="(\\b${suite}/[a-zA-Z0-9]+\\.feature:[0-9]+)"
     if [[ " ${AVAILABLE_SUITES[*]} " != *" $suite "* ]]; then
-      pattern="(${suite}/[a-zA-Z0-9]+\\.feature:[0-9]+)"
       NONEXISTING_SCENARIOS+=($(grep -Eo ${pattern} ${PATH_TO_EXPECTED_FAILURE_FILE}))
+    else
+      SCENARIOS=($(grep -Eo ${pattern} ${PATH_TO_EXPECTED_FAILURE_FILE} | grep -Eo "[a-zA-Z0-9]+\\.feature:[0-9]+"))
+      for scenario in "${SCENARIOS[@]}"; do
+        FEATURE_FILE=$(echo "$scenario" | cut -d':' -f1)
+        if [[ ! -f "$PATH_TO_SUITES/$suite/$FEATURE_FILE" ]]; then
+          NONEXISTING_SCENARIOS+=("$suite/$scenario")
```
Comment (amrita-shrestha) [inline tests/acceptance/check-deleted-suites-in-expected-failure.sh:55]: only for readability
```suggestion
          NON_EXISTING_SCENARIOS+=("$suite/$scenario")
```

File: tests/acceptance/check-deleted-suites-in-expected-failure.sh [hunk 2]
```diff
@@ -42,20 +42,28 @@ for expected_failure_file in "${EXPECTED_FAILURE_FILES[@]}"; do
   # also filter the duplicated suites name
   EXPECTED_FAILURE_SUITES=($(echo "${EXPECTED_FAILURE_SUITES[@]}" | tr ' ' '\n' | sort | uniq))
   # Check the existence of the suite
-  NONEXISTING_SCENARIOS=()
+  NON_EXISTING_SCENARIOS=()
   for suite in "${EXPECTED_FAILURE_SUITES[@]}"; do
```
Comment (saw-jan) [inline tests/acceptance/check-deleted-suites-in-expected-failure.sh:46]:
```suggestion
  NON_EXISTING_SCENARIOS=()
  FEATURE_PATTERN="[a-zA-Z0-9]+\\.feature:[0-9]+"
  for suite in "${EXPECTED_FAILURE_SUITES[@]}"; do
```

File: tests/acceptance/check-deleted-suites-in-expected-failure.sh [hunk 3]
```diff
@@ -42,20 +42,28 @@ for expected_failure_file in "${EXPECTED_FAILURE_FILES[@]}"; do
...
+    pattern="(\\b${suite}/[a-zA-Z0-9]+\\.feature:[0-9]+)"
```
Comment (saw-jan) [inline tests/acceptance/check-deleted-suites-in-expected-failure.sh:47]:
```suggestion
    pattern="(\\b${suite}/${FEATURE_PATTERN}\\b)"
```

File: tests/acceptance/check-deleted-suites-in-expected-failure.sh [hunk 4]
```diff
@@ -42,20 +42,28 @@ for expected_failure_file in "${EXPECTED_FAILURE_FILES[@]}"; do
...
     if [[ " ${AVAILABLE_SUITES[*]} " != *" $suite "* ]]; then
-      pattern="(${suite}/[a-zA-Z0-9]+\\.feature:[0-9]+)"
-      NONEXISTING_SCENARIOS+=($(grep -Eo ${pattern} ${PATH_TO_EXPECTED_FAILURE_FILE}))
+      NON_EXISTING_SCENARIOS+=($(grep -Eo ${pattern} ${PATH_TO_EXPECTED_FAILURE_FILE}))
+    else
+      SCENARIOS=($(grep -Eo ${pattern} ${PATH_TO_EXPECTED_FAILURE_FILE} | grep -Eo "[a-zA-Z0-9]+\\.feature:[0-9]+"))
```
Comment (saw-jan) [inline tests/acceptance/check-deleted-suites-in-expected-failure.sh:51]:
```suggestion
      SCENARIOS=($(grep -Eo ${pattern} ${PATH_TO_EXPECTED_FAILURE_FILE} | grep -Eo "${FEATURE_PATTERN}"))
```

**Review by saw-jan — APPROVED** [review]

**Review by amrita-shrestha — APPROVED** [review]

**Pull request merged**